File: backend/src/Salmandyar.Domain/Entities/Assessments/AssessmentAssignment.cs
using Salmandyar.Domain.Enums;

namespace Salmandyar.Domain.Entities.Assessments;

public class AssessmentAssignment
{
    public int Id { get; set; }

    public string UserId { get; set; } = string.Empty;
    public virtual User User { get; set; } = null!;

    public int FormId { get; set; }
    public virtual AssessmentForm Form { get; set; } = null!;

    public DateTime AssignedDate { get; set; } = DateTime.UtcNow;
    public DateTime? Deadline { get; set; }
    public bool IsMandatory { get; set; } = false;

    public AssessmentAssignmentStatus Status { get; set; } = AssessmentAssignmentStatus.Pending;

    // Link to the submission if completed
    public int? SubmissionId { get; set; }
    public virtual AssessmentSubmission? Submission { get; set; }

    public DateTime? CompletedDate { get; set; }
}

File: frontend/src/types/assessment-assignment.ts
export enum AssessmentAssignmentStatus {
    Pending = 0,
    InProgress = 1,
    Completed = 2,
    Expired = 3,
    Cancelled = 4
}

export interface AssessmentAssignment {
    id: number;
    userId: string;
    userFullName: string;
    formId: number;
    formTitle: string;
    assignedDate: string;
    deadline?: string;
    isMandatory: boolean;
    status: AssessmentAssignmentStatus;
    submissionId?: number;
    score?: number;
    completedDate?: string;
    submissionDetails?: SubmissionDetail;
}

export interface SubmissionDetail {
    id: number;
    totalScore: number;
    submittedAt: string;
    answers: QuestionAnswerDetail[];
}

export interface QuestionAnswerDetail {
    questionId: number;
    questionText: string;
    weight: number;
    tags: string[];
    selectedOptionText?: string;
    selectedOptionScore?: number;
    textResponse?: string;
    booleanResponse?: boolean;
}

export interface CreateAssessmentAssignment {
    userId: string;
    formId: number;
    deadline?: string;
    startDate?: string;
    isMandatory: boolean;
}

export interface UserAssessmentSummary {
    userId: string;
    fullName: string;
    role: string;
    isActive: boolean;
    totalAssigned: number;
    completed: number;
    pending: number;
    overdue: number;
}
